Admin/MigrateDLs.ps1

```diff
@@ -0,0 +1,347 @@
+    - If validation is successful, we will remove the DL from on-premises (check if the object was removed) and create a mail-contact on-premises, having the ExternalEmailAddress pointing to the provisioned onmicrosoft.com email address of the DL on EXO.
+    - Once the Mail-Contact has been validated, we will rename the log file to start with 'OK-'
+    - If the process fails on any step, we will rename the log file to start with 'Failed-'
+
+    PRE-REQUISITES:
+    For the process to work correctly, we need to have the following things in consideration:
+    - Make sure the DL's you include on the CSV file are present in Exchange On-premises AND NOT SYNCED to Exchange Online anymore (in other words, the DL should not be present in EXO)
+    - Running this script, needs to be done from the Exchange Management Shell on-premises, with the Exchange Online management module installed (if you need to install the module, running EMS or Powershell as an admin, run 'Install-Module ExchangeOnlineManagement')
+    - The account running this scripts needs to have at least the 'Recipient Management' Exchange management role assigned on both ends (on-premises and online).
+    - The CSV file needs to have just one column named 'DL'
+    - The specified log path should not end with '\'
+    - To avoid any possible throttling applied to Powershell, I recommend running batches of no more than 100 entries per day.
+
+.PARAMETER CSVfile
+    CSV file of the DL's we will be migrating when running this script
+
+.PARAMETER LogPath
+	Temp path for storing DL props and logging of the process for each one of the entries listed on the CSV file
+
+.EXAMPLE
+    .\MigrateDLs.ps1 -CSV c:\Temp\myCSV.csv -LogPath C:\Temp
+    This will import the specified CSV file and migrate the DL's listed there. For every one of them, their properties and the process details will be stores ont he specified path for logging.
+#>
+
+
+param(
+    [Parameter(Mandatory = $True)]
+    [System.String]$CSV,
+    [Parameter(Mandatory = $True)]
+    [System.String]$LogPath
+)
+
+. $PSScriptRoot\..\Shared\OutputOverrides\Write-Host.ps1
+
+if (!(Test-Path -Path $LogPath )) {
+    Write-Host "  Creating Directory: $LogPath"
+    New-Item -ItemType directory -Path $LogPath | Out-Null
+}
+
+if (($LogPath).EndsWith('\')) {
+    Write-Host "ERROR: The specified LogPath is ending with a '\'. This should not be present and should be something like 'C:\temp' instead of 'C:\temp\'" -ForegroundColor Red
+    exit
+}
+
+Write-Host ""
+Write-Host "### Starting migrate DL's to EXO script ###"
+Write-Host "-------------------------------------------"
+Write-Host "  Importing the CSV that contains the DL's to migrate from $CSV"
+$Objects = Import-Csv $CSV
+if (!($Objects).DL) {
+    Write-Host "  ERROR: The specified CSV file doesn't contain a column named 'DL'. Please specify a valid CSV file that contains the mentioned column. Exiting the process." -ForegroundColor Red
```

Comment:
Why would we use a CSV where the DG is identified by a "DL" column? This doesn't make much sense, because there's no property of a DL called "DL". For example, to make your CSV you could do something like:

Get-DistributionGroup | Export-Csv .\some.csv

You would end up with columns like DistinguishedName and Identity which you could use to find the DL.

Also, we should consider not expecting a CSV at all, but just taking pipeline input. That way, the user could either run the script live, like:

Get-DistributionList | .\MigrateDLs.ps1

Or the user could create a CSV, edit it, and then pipe that:

Import-Csv .\my.csv | .\MigrateDLs.ps1